### PR TITLE
feature(apphealth&configuration): add app health check & adapt new configuration API

### PR DESCRIPTION
**Pull request by 1046102779**
There are three changes:
1. Add app health check.
2. Change Configuration Get&Subscribe API.
3. remove dependency `github.com/dapr/dapr` and copy `dapr/proto`  from `github.com/dapr/dapr` to `github.com/dapr/go-sdk` .

/cc @daixiang0 @yaron2 @ItalyPaleAle

### CODE REVIEW

**Comment by 1046102779** [comment]
All checks have passed. 

image

**Comment by yaron2** [comment]
@1046102779 did you run all the example apps to make sure there are no regressions?

File: client/actor.go [hunk 1]
```diff
@@ -22,10 +22,10 @@ import (
 	anypb "github.com/golang/protobuf/ptypes/any"
 	"github.com/pkg/errors"
 
-	pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
 	"github.com/dapr/go-sdk/actor"
 	"github.com/dapr/go-sdk/actor/codec"
 	"github.com/dapr/go-sdk/actor/config"
+	pb "github.com/dapr/go-sdk/dapr/proto/runtime/v1"
```
Comment (yaron2) [inline client/actor.go:28]: Why not use `dapr/dapr`'s proto package?

Comment (1046102779) [reply client/actor.go:28]: It is too heavy for dapr `go-sdk` to directly depend on the `dapr/dapr` repo. I suggest that it is best to separate proto into the standard repo that named `protocol`.

At present,  go-sdk directly imports the `dapr/dapr` module, which will cause the micrologic service to import `dapr/dapr` in go.mod. It is appropriate for the business side to only introduce APIs.

1. Related issue: https://github.com/dapr/dapr/issues/4920
2. Related PR: https://github.com/dapr/dapr/pull/4923

Comment (1046102779) [reply client/actor.go:28]: 1. 
https://github.com/dapr/go-sdk/tree/main/dapr/proto

2. 
image

In the previous go-sdk, the introduction of dapr proto was confusing. It not only directly imported `dapr/dapr` proto, but also directly copied proto to `github.com/dapr/go-sdk/dapr/proto`.

So i removed `dapr/dapr` dependency and copy proto to this repo.

**Comment by 1046102779** [comment]
> @1046102779 did you run all the example apps to make sure there are no regressions?

Currently only compiled and built, there is no problem with this.  All examples are not validated locally.

**Comment by berndverst** [comment]
We need this change in the Go SDK -- or anything that updates the Configuration API response.

Currently our certification tests have pinned a random go-sdk fork from user hunter007.

@yaron2

**Review by yaron2 — APPROVED** [review]

**Pull request merged**